clang/test/SemaSYCL/unnamed-kernel.cpp

```diff
@@ -117,8 +113,7 @@ struct MyWrapper {
 int main() {
   cl::sycl::queue q;
 #ifndef __SYCL_UNNAMED_LAMBDA__
-  // expected-error-re@#KernelSingleTask {{'(lambda at {{.*}}unnamed-kernel.cpp{{.*}}' is an invalid kernel name type}}
-  // expected-note@#KernelSingleTask {{unnamed type used in a SYCL kernel name}}
+  // expected-error-re@#KernelSingleTask {{unnamed lambda '(lambda at {{.*}}unnamed-kernel.cpp{{.*}}' used}}
```

Comment:
@AaronBallman, if we want to retain the filename for this diagnostic check, is there a way to do it without spelling out the name?

Reply:
You can't really save the filename without losing the ability to do the named-diagnostics.  IMO, the named diagnostics are superior and we'll need to remember over time that they are in sycl.hpp.

Reply:
@erichkeane, I am not sure we are talking about the same name.   I believe you are talking about the "#KernelSingleTask"; I like that too.  I am asking about the 'unnamed-kernel.cpp' usage within the diagnostic; if there is a way to reg-exp that.

Reply:
Ah, you're right, we ARE referring to different things :)  

My only thought is PERHAPS some use of the __FILE__ macro?  But that would be difficult thanks to needing to strip the directory?  But even that I'm not sure of.  I don't think there is a way of doing that, and VerifyDiagnosticsConsumer doesn't have any special feature for that.

Reply:
Yeah, I don't know of an easy way to do that (and I suspect the `__FILE__` macro behavior may depend on what Clang is trying to be compatible with at the moment, so it may vary between targets). I think what we have here is pretty reasonable though.